### ISSUE TITLE
Routing inconsistencies

### ISSUE THREAD

**Issue by suchita-jain** [istio/istio#23161]
**Background**
We have an EKS cluster with Istio app mesh with around 30 services with around 30k TPS.

**Bug description**
We had a robot attack increasing traffic from 34k to 57k resulting in autoscaling from 60 to 104 nodes with pilot scaling up from 10 to max 50 pods and ingress-gateway scaling up from 50 to 461 pods. The attack was mitigated within 5 minutes. The descaling began shortly after. But the routing got inconsistent. Ingress gateway started to route to old pods IPs that were not present or were being replaced. We started receiving around 5k 503 UF,URX errors after the attack mitigation for several hours. The issue was mitigated by restarting all the ingress-gateway and pilot pods.  

pilot log
```
error   Endpoint without pod 10.0.9.129 xxx.default

warn    failed to get pod default/xxx from kube-apiserver: pods "xxx" not found
``` 

**Expected behavior**
The routing should not have gotten inconsistent. 

**Steps to reproduce the bug**
Haven't tried.

**Version**
Istio: 1.4.5
EKS: 1.14

**Comment by howardjohn** [comment]
related? https://github.com/istio/istio/issues/23118

**Comment by suchita-jain** [comment]
Yes, same issue.

**Comment by harshal-shah** [comment]
We rolled back our Istio from `1.4.7` to `1.4.3` and have not faced the `pods "abc-5674bc9fb7-mxpzt" not found` error since then.

**Comment by crhuber** [comment]
https://github.com/istio/istio/issues/23118 for us was also caused during a period of a high amount of churn. We have a deployment that creates roughly 300 new pods and during this time is when we see this error

**Comment by suchita-jain** [comment]
@crhuber, Was the error recurring in your case? We faced it only once and haven't been able to reproduce it. Were you able to reproduce the issue? If yes, what are the steps? Was the Pilot or any other system component under-resourced at that time?

**Comment by harshal-shah** [comment]
@suchita-jain we had this happen 2-3 times since the upgrade. We have not had this error again since we downgraded to 1.4.3 and that makes sense because the code-path which introduced this error was added in [1.4.4](https://istio.io/news/releases/1.4.x/announcing-1.4.4/) 
PR with the change https://github.com/istio/istio/pull/19035

**Comment by suchita-jain** [comment]
Thanks, @harshal-shah! It was not introduced in 1.4.4 but wasn't fixed in it due to a merge conflict. Seems like it was fixed in 1.4.5: https://github.com/istio/istio/issues/20062 We are using 1.4.5.  But still faced the issue. Did you change anything else while downgrading the Istio version?

**Comment by howardjohn** [comment]
I am nearly certain this is fixed by https://github.com/istio/istio/issues/25112. Lets move the conversation there, in particular if the fixed provided there does not solve this. Thanks!
